rust/src/operations/optimize.rs

```diff
@@ -1180,11 +1180,11 @@ pub(super) mod zorder {
             .map(|i| (i * value_size) as i32)
             .collect::<Vec<i32>>();
 
-        let out_arr = BinaryArray::new_unchecked(
+        let out_arr = BinaryArray::try_new(
```

Comment:
`new_unchecked` was made unsafe. This does the same thing, just with some validation checks. I'm not aware of any significant performance hits with this extra check.